src/components/Admin/RentalData/index.js

```diff
@@ -14,35 +15,105 @@ import * as S from './RentalData.style';
 // Table props
 import rentalDataColumns from './rentalDataColumns';
 
+// custom hooks
+import useFetch from '../../../hooks/useFetch';
+
+// constants
+import { statusEnum } from '../../../constants/rentalRecords';
+
 // routes
 import { routes } from '../../../constants/adminRoutes';
 import { ADD_RENTAL_URL } from '../../../constants/navRoutes';
 
-// custom hooks
-import useFetch from '../../../hooks/useFetch';
-
 const { RENTAL_DATA_ALL, RENTAL_DATA_SINGLE } = routes;
 
-function RentalData() {
+const { Option } = Select;
+
+function RentalData({ history }) {
+  const [modalVisible, setModalVisible] = useState(false);
+  const [editingStatus, setEditingStatus] = useState(null);
+  const [updatingRecord, setUpdatingRecord] = useState(false);
+  const [recordToUpdate, setRecordToUpdate] = useState(null);
+  const [rentalData, setRentalData] = useState(null);
+  const [loading, setLoading] = useState(false);
+
+  const toggleModal = () => setModalVisible(!modalVisible);
+
+  const editStatus = bool => {
+    setEditingStatus(bool);
+  };
+
+  const changeStatus = status => {
+    const { rentalId } = recordToUpdate;
+    setRecordToUpdate({ rentalId, updateType: status });
+  };
+
+  const updateRecord = (rentalId, updateType) => {
+    setRecordToUpdate({ rentalId, updateType });
+    toggleModal();
+  };
+
+  const confirmUpdate = async () => {
+    const { rentalId, updateType } = recordToUpdate;
+    let response;
+    try {
+      setUpdatingRecord(true);
+      if (updateType === 'delete') {
+        response = await axios.delete('/.netlify/functions/deleteRecord', {
+          data: { rentalId },
+        });
+      } else if (statusEnum.includes(updateType)) {
+        response = await axios.patch('/.netlify/functions/setRentalStatus', {
+          rentalId,
+          newStatus: updateType,
+        });
+      }
+      const { data: newRentalData } = await axios.get(
+        '/.netlify/functions/getRentalData'
+      );
```

Comment:
Here as well you can update the current data based on the condition and it will reflect the new table instead of sending new requests.